src/crush/CrushWrapper.cc

```diff
@@ -1647,12 +1647,22 @@ int CrushWrapper::update_device_class(int id,
                                       const string& name,
                                       ostream *ss)
 {
+  assert(item_exists(id));
+  auto old_class_name = get_item_class(id);
+  stringstream ts;
+  if (old_class_name &&
+      old_class_name != class_name && // so command is idempotent
+      class_is_in_use(get_class_id(old_class_name), &ts)) {
+    *ss << "osd." << id << " has already bound to class '" << old_class_name
+        << "', which is currently " << ts.str();
+    return -EBUSY;
```

Comment:
> There has to be something the user can do if the osd is misclassified as hdd vs ssd and there are rules for both.

How about dropping the constranits above and warning about potential data losing (and migrating) if any,  and requiring user to pass "--yes-i-really-mean-it" to go on then?



> Maybe we should leave class create and rm

But an empty class is meanless unless it is bounded to particular devices.